src/lakefs_spec/spec.py

```diff
@@ -397,6 +562,29 @@ def put_file(
         storage_options: dict[str, Any] | None = None,
         **kwargs: Any,
     ) -> None:
+        """
+        Upload a local file to a remote location on a lakeFS server.
+
+        Parameters
+        ----------
+        lpath: str
+            The local path on disk to upload to the lakeFS server.
+        rpath: str
+            The remote target path to upload the local file to. Must be a fully qualified lakeFS URI.
+        callback: fsspec.callbacks.Callback
+            An fsspec callback to use during the operation. Can be used to report download progress.
+        precheck: bool
+            Check if ``lpath`` already exists and compare its checksum with that of ``rpath``, skipping the download if they match.
+        use_blockstore: bool
+            Optionally upload the file directly to the underlying block storage, thereby bypassing the lakeFS server and saving a
+            file transfer. Preferable for uploads of large files.
+        presign: bool
+            Whether to use pre-signed URLs to upload the object if ``use_blockstore == True``.
```

Comment:
See above; mention the implications if `use_blockstore=True` and `presign=False`